### PR TITLE
Ccsid override

### PR DESCRIPTION
**Pull request by msillence**
Suspect I need to do a little more work on this yet but the tests are starting to show promise

### CODE REVIEW

File: journal-parsing/src/main/java/com/fnz/db2/journal/retrieve/JdbcFileDecoder.java [hunk 1]
```diff
@@ -278,10 +342,10 @@ public static AS400DataType toDataType(String name, String type, int length, Int
 				return new AS400VarBin(length);
 			case "XML":
 				return AS400_XML;
-//			case "CLOB":
-//				return new AS400Clob(as400);
+//			case "CLOB": 
+//				return new AS400Clob(as400); 
 			default:
-				Matcher matcher = BIT_DATA.matcher(type);
+				Matcher matcher = BIT_DATA.matcher(type); //  - treat as binary
```
Comment (msillence) [inline journal-parsing/src/main/java/com/fnz/db2/journal/retrieve/JdbcFileDecoder.java:348]: I think these should be AS400ByteArray?

File: jt400-override-ccsid/readme.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+
+# Purpose
+
+Override the ccsid and force the character encoding
+
+really you should not need this and just set the right encoding on the tables sadly if you have a large legacy code base and a large number of customers who have this incorrectly configured that correct solution might be costly and risky
```
Comment (martintosney) [inline jt400-override-ccsid/readme.md:6]: It's not so much having the wrong CCSID on the tables that is the problem. The system can transparently handle conversions between the encodings and make this work, provided the characters are present in both code pages.

The issue is specifically with a [system value called QCCSID](https://www.ibm.com/docs/en/i/7.5?topic=faqs-i-system-value-qccsid). If this is set to 65535 (binary), it indicates that jobs will have no CCSID, and no conversion will be performed.

File: debezium-connector-ibmi/src/main/java/io/debezium/connector/db2as400/As400JdbcConnection.java [hunk 3]
```diff
@@ -40,6 +41,8 @@ public class As400JdbcConnection extends JdbcConnection implements Connect<Conne
     private static final String URL_PATTERN = "jdbc:as400://${" + JdbcConfiguration.HOSTNAME + "}/${"
             + JdbcConfiguration.DATABASE + "}";
     private final JdbcConfiguration config;
+    private final int forcedCcsid;
+    private boolean registerd = false;
```
Comment (martintosney) [inline debezium-connector-ibmi/src/main/java/io/debezium/connector/db2as400/As400JdbcConnection.java:45]: Spelling - "registered"

**Review by martintosney — APPROVED** [review]

File: jt400-override-ccsid/readme.md (hunk 2 above)
Comment (martintosney) [reply jt400-override-ccsid/readme.md:6]: Actually, here's a slightly better description of the problem

https://www.ibm.com/support/pages/what-impact-changing-qccsid-shipped-65535-another-ccsid

**Pull request merged**